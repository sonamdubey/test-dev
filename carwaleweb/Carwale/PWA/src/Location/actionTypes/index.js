export const OPEN_POPUP = 'OPEN_POPUP';
export const CLOSE_POPUP = 'CLOSE_POPUP';
export const SHOW_CROSS_ICON = 'SHOW_CROSS_ICON';
export const HIDE_CROSS_ICON = 'HIDE_CROSS_ICON';
