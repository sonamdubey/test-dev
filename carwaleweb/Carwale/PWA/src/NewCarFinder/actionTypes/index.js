export const NEXT_FILTER = 'NEXT_FILTER'
export const SET_CURRENT_SCREEN = 'SET_CURRENT_SCREEN'
export const SET_ALL_FILTER_SCREEN = 'SET_ALL_FILTER_SCREEN'

export const REQUEST_CAR_FEATURE = 'REQUEST_CAR_FEATURE'
export const RECEIVE_CAR_FEATURE = 'RECEIVE_CAR_FEATURE'
export const CAR_FEATURE_SELECTION = 'CAR_FEATURE_SELECTION'

// budget
export const UPDATE_BUDGET_SLIDER_VALUE = 'UPDATE_BUDGET_SLIDER_VALUE'
export const UPDATE_BUDGET_INPUT_VALUE = 'UPDATE_BUDGET_INPUT_VALUE'

// body type
export const REQUEST_BODY_TYPE = 'REQUEST_BODY_TYPE'
export const RECEIVE_BODY_TYPE = 'RECEIVE_BODY_TYPE'
export const BODY_TYPE_SELECTION = 'BODY_TYPE_SELECTION'
export const SET_BODY_STORED_FILTERS = 'SET_BODY_STORED_FILTERS'


// fuel type
export const REQUEST_FUEL_TYPE = 'REQUEST_FUEL_TYPE'
export const RECEIVE_FUEL_TYPE = 'RECEIVE_FUEL_TYPE'
export const FUEL_TYPE_SELECTION = 'FUEL_TYPE_SELECTION'
export const SET_FUEL_STORED_FILTERS = 'SET_FUEL_STORED_FILTERS'

// fuel type
export const REQUEST_MAKE = 'REQUEST_MAKE'
export const RECEIVE_MAKE = 'RECEIVE_MAKE'
export const MAKE_SELECTION = 'MAKE_SELECTION'
export const SET_MAKE_STORED_FILTERS = 'SET_MAKE_STORED_FILTERS'



// listing
// model list RECEIVE_MODEL_LIST
export const REQUEST_MODEL_LIST = 'REQUEST_MODEL_LIST'
export const RECEIVE_MODEL_LIST = 'RECEIVE_MODEL_LIST'
export const APPEND_MODEL_LIST = 'APPEND_MODEL_LIST'

// filter screen
export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'

// compare cars
export const ADD_MODEL_TO_COMPARE = 'ADD_MODEL_TO_COMPARE'
export const REMOVE_MODEL_FROM_COMPARE = 'REMOVE_MODEL_FROM_COMPARE'
// header
export const MAKE_HEADER_STICKY = 'MAKE_HEADER_STICKY'
export const MAKE_HEADER_NORMAL = 'MAKE_HEADER_NORMAL'
