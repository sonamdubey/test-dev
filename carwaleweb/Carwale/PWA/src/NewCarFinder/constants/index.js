/**
 * API URL's
 */
export const SUGGESTED_BUDGET_ENDPOINT = "/api/newcarfinder/budget/"
export const NEWCAR_SEARCH_ENDPOINT = "/api/newcarfinder/search/"
export const SUGGESTED_BODYTYPE_ENDPOINT = "/api/newcarfinder/bodytype/"
export const FUELTYPE_ENDPOINT = "/api/newcarfinder/fueltype/"
export const MAKE_ENDPOINT = "/api/newcarfinder/filters/make/"
export const VERSION_CITY_PRICE_ENDPOINT = "/api/prices/version/"
export const DEFAULT_FILTERS_ENDPOINT = "/api/filters/all/"
export const FILTER_SCREEN_ENDPOINT = "/api/newcarfinder/screen/"

/**
 * PAGE URL's
 */
export const NEWCARFINDER_ENDPOINT = '/find-car/'
export const NEWCARFINDER_FILTERS_ENDPOINT = '/find-car/filters/'
export const NEWCARFINDER_RESULTS_ENDPOINT = '/find-car/results/'
export const NEWCARFINDER_FILTERS_BODYTYPE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"body-type/"
export const NEWCARFINDER_FILTERS_FUELTYPE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"fuel-type/"
export const NEWCARFINDER_FILTERS_MAKE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"make/"
export const NEWCARFINDER_FILTERS_BUDGET_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"budget/"
/**
 * SEO Pages URL's
 */
export const NEWCARFINDER_SEO_SINGLE_DIMENSION_ENDPOINT = '/best-([a-zA-Z]+)-cars-in-india/'

/**
 * NCF CONSTANTS
*/
export const BUDGET_SLIDER_MIN = 200000
export const BUDGET_SLIDER_MAX = 10000000
export const CATEGORY_NAME = "NCFPWA"
export const PAGE_VIEW_CATEGORY = "PageViews"
export const SEO_TYPE = {
    SEO_SINGLE: 'SEO_SINGLE'
}
