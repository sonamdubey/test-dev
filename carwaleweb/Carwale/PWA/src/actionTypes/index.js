// location: city
export const SELECT_CITY = "SELECT_CITY";
export const SELECT_AREA = "SELECT_AREA";
export const SET_LOCATION = "SET_LOCATION";
export const PREFILL_LOCATION = "PREFILL_LOCATION";

// toast
export const SET_TOAST = "SET_TOAST";
export const REMOVE_TOAST = "REMOVE_TOAST";

// header
export const SHOW_HEADER = "SHOW_HEADER";
export const HIDE_HEADER = "HIDE_HEADER";

// footer
export const SHOW_FOOTER = "SHOW_FOOTER";
export const HIDE_FOOTER = "HIDE_FOOTER";

// navigation drawer
export const OPEN_NAV_DRAWER = "OPEN_NAV_DRAWER";
export const CLOSE_NAV_DRAWER = "CLOSE_NAV_DRAWER";

// fullscreen
export const HIDE_HEADER_FOOTER = "HIDE_HEADER_FOOTER";
export const SHOW_HEADER_FOOTER = "SHOW_HEADER_FOOTER";

//page
export const SET_PAGE = "SET_PAGE";

// autoDetect location
export const AUTODETECT_LOCATION = "AUTODETECT_LOCATION"

export const SELECT_LATLONG = "SELECT_LATLONG";
export const FETCHING_LOCATION = "FETCHING_LOCATION"
export const SELECT_AREA_CITIES = "SELECT_AREA_CITIES"
