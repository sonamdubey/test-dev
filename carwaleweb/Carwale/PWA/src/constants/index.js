/**
 * API URL's
 */
export const CITY_AUTOCOMPLETE_URL = "/api/v2/autocomplete/city/"
export const AREA_AUTOCOMPLETE_URL = "/api/v2/autocomplete/areas/"


/**
 * UNIQUE ID's
 */
export const UNIVERSAL_ANALYTICS_ID = "UA-337359-1"


/**
 * App download links
 */
export const APP_STORE_LINK = 'https://itunes.apple.com/in/app/carwale/id910137745?mt=8'
export const PLAY_STORE_FOOTER_LINK = 'https://play.google.com/store/apps/details?id=com.carwale&referrer=utm_source%3DCarWaleMsite%26utm_medium%3DFooter%26utm_campaign=CarWale%2520MobilesiteFooter'
export const PLAY_STORE_NAV_DRAWER_LINK = 'https://play.google.com/store/apps/details?id=com.carwale&referrer=utm_source%3DCarWaleMsite%26utm_medium%3DMenuDrawer%26utm_campaign=CarWale%2520MobilesiteMenuDrawer'


export const desktopBreakPoint = 780
