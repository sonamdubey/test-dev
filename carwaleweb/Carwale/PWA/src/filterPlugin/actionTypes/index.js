export const OPEN_FILTER_SCREEN = 'OPEN_FILTER_SCREEN'
export const CLOSE_FILTER_SCREEN = 'CLOSE_FILTER_SCREEN'
export const SET_FILTERS = 'SET_FILTERS'
