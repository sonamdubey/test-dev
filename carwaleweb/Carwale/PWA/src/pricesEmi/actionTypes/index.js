// budget
export const UPDATE_MODEL_DATA= 'UPDATE_MODEL_DATA'
export const UPDATE_DOWNPAYMENT_SLIDER_VALUE = 'UPDATE_DOWNPAYMENT_SLIDER_VALUE'
export const UPDATE_DOWNPAYMENT_INPUT_VALUE = 'UPDATE_DOWNPAYMENT_INPUT_VALUE'
export const UPDATE_TENURE_SLIDER_VALUE = 'UPDATE_TENURE_SLIDER_VALUE'
export const UPDATE_TENURE_INPUT_VALUE = 'UPDATE_TENURE_INPUT_VALUE'
export const UPDATE_INTEREST_SLIDER_VALUE = 'UPDATE_INTEREST_SLIDER_VALUE'
export const UPDATE_INTEREST_INPUT_VALUE = 'UPDATE_INTEREST_INPUT_VALUE'
export const UPDATE_PIE_VALUE = 'UPDATE_PIE_VALUE'
export const SHOW_EMI_POPUP = 'SHOW_EMI_POPUP'
export const HIDE_EMI_POPUP = 'HIDE_EMI_POPUP'

// set model id for emi
export const SET_EMI_MODEL = 'SET_EMI_MODEL'
export const UPDATE_EMI_MODEL = 'UPDATE_EMI_MODEL'
export const SET_DOWNPAYMENT_SNAPPOINT = 'SET_DOWNPAYMENT_SNAPPOINT'
export const SET_IS_EMI_VALID = 'SET_IS_EMI_VALID'
