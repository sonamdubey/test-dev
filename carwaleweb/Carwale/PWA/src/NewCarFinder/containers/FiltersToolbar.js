import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import {CATEGORY_NAME} from '../constants/index';
import {
	createRipple
} from '../../utils/Ripple'

import {
	openFilterScreen
} from '../actionCreators/FiltersScreen'


import { throttle } from 'throttle-debounce'
import { fireInteractiveTracking } from '../../utils/Analytics';
import { trackCustomData } from '../../utils/cwTrackingPwa';
/**
 * This class representing the container component
 * for filters toolbar
 *
 * @class FiltersToolbar
 * @extends {React.Component}
 */
class FiltersToolbar extends React.Component {
	constructor(props) {
		super(props)

		this.lastScrollPosition = 0
	}

	componentDidMount() {
		//window.addEventListener('scroll', this.floatToolbar)
	}

	componentWillUnmount() {
		//window.removeEventListener('scroll', this.floatToolbar)
	}

	floatToolbar = () => {
		let windowScrollPosition = window.pageYOffset || document.documentElement.scrollTop

		this.placeholderOffsetTop = this.refs.toolbarPlaceholder.getBoundingClientRect().top + windowScrollPosition

		if (this.lastScrollPosition > windowScrollPosition) {
			this.refs.toolbarContent.classList.add('toolbar--fixed')

			if (windowScrollPosition < this.placeholderOffsetTop) {
				this.refs.toolbarContent.classList.remove('toolbar--fixed')
			}
		}
		else {
			this.refs.toolbarContent.classList.remove('toolbar--fixed')
		}

		this.lastScrollPosition = windowScrollPosition
	}

	handleEditClick = (event) => {
		this.props.openFilterScreen()
		this.trackEvent("EditOptionClick")
		trackCustomData(CATEGORY_NAME,"ListingEditOptionClick","filterType=none",false)
	}

	handleFilterTabClick = (event) => {
		createRipple(event)
		const filterType = event.currentTarget.getAttribute('data-filter-type')

		this.props.openFilterScreen(filterType)
		this.trackEvent("EditOptionClick_" + filterType)
		trackCustomData(CATEGORY_NAME,"ListingEditOptionClick","filterType="+filterType,false)
	}

	getSelections = () => {
		const { selectionObject } = this.props
		if (selectionObject) {
			return Object.keys(selectionObject).map(selection => {
				return (<li className="filter-toolbar-list__item" key={selection}>
					<div
						className="btn-tab"
						data-filter-type={selection}
						onClick={this.handleFilterTabClick}
					>
						<span className="tab__key">{selection}</span>
						<span className="tab__value">{selectionObject[selection]}</span>
					</div>
				</li>)
			});
		}
	}

	trackEvent(label) {
		fireInteractiveTracking("NCFPWA", "NCF_Listing", label)
	}

	render() {
		return (
			<div ref="toolbarPlaceholder" className="filters__toolbar-placeholder">
				<div ref="toolbarContent" className="filters-toolbar__content">
					<div className="toolbar--relative">
						<div className="filters-toolbar-list__content">
							<ul className="filter-toolbar__list">
								{this.getSelections()}
							</ul>
						</div>
						<div
							onClick={this.handleEditClick}
							className="filter__edit"
						>
							Edit
						</div>
					</div>
				</div>
			</div>
		)
	}
}

const mapDispatchToProps = (dispatch, getState) => {
	return {
		openFilterScreen: bindActionCreators(openFilterScreen, dispatch)
	}
}

export default connect(null, mapDispatchToProps)(FiltersToolbar)
