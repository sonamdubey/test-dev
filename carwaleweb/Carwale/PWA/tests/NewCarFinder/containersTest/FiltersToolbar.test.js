import React from 'react'
import { Redirect } from 'react-router-dom'
import FiltersToolbar from '../../../src/NewCarFinder/containers/FiltersToolbar'
import { NEWCARFINDER_RESULTS_ENDPOINT } from '../../../src/NewCarFinder/constants/index'
import {OPEN_FILTER_SCREEN} from '../../../src/NewCarFinder/actionTypes'

describe('Tests for FiltersToolbar component', () => {
    let state
    let props
    let store
    let enzymeWrapper

    const Wrapper = () => {
        if(!enzymeWrapper){
            store = mockStore(state)
            enzymeWrapper = shallowWithStore(<FiltersToolbar {...props}/>, store)

        }
        return enzymeWrapper
    }

    beforeEach( () => {
        props = undefined
        state = undefined
        enzymeWrapper = undefined
        store = undefined
    })

    it('should render snapshot', () => {
        props = {
            selectionObject:{
                'Budget' : '₹ 5 lakh',
                'Body Type':'Sedan, Coupe',
                'Fuel Type':'Diesel, Petrol'
            }
        }
        const wrapper = Wrapper().dive()
        expect(wrapper).toMatchSnapshot()
    });

    describe('on clicking edit button', () => {

        //https://github.com/airbnb/enzyme/issues/697
        //https://github.com/airbnb/enzyme/issues/208
        //This test requires more research,
        //the methods suggested in above issues did not prove useful
        //https://github.com/airbnb/enzyme/issues/365
        // it('should call handleEditClick',() => {
        //     props = {
        //         selectionObject:{
        //             'Budget' : '₹ 5 lakh',
        //             'Body Type':'Sedan, Coupe',
        //             'Fuel Type':'Diesel, Petrol'
        //         },
        //         openFilterScreen : jest.fn()
        //     }

        //     const handleEditClick = jest.fn()
        //     //const oldhandleEditClick = FiltersToolbar.WrappedComponent.prototype.handleEditClick
        //     //FiltersToolbar.WrappedComponent.prototype.handleEditClick = handleEditClick
        //     const wrapper = shallow(<FiltersToolbar.WrappedComponent {...props}/>)
        //     console.log(wrapper.instance())
        //     wrapper.instance().handleEditClick = handleEditClick
        //     wrapper.update()
        //     const editButton = wrapper.find('.filter__edit')
        //     editButton.simulate('click')
        //     expect(handleEditClick).toBeCalled()
        //     //FiltersToolbar.WrappedComponent.prototype.handleEditClick = oldhandleEditClick
        // })

        it('should call openFilterScreen and trackEvent',() => {
            const wrapper = Wrapper().dive()
            wrapper.instance().trackEvent = jest.fn()

            const editButton = wrapper.find('.filter__edit')
            editButton.simulate('click')
            expect(wrapper.instance().trackEvent).toBeCalledWith('EditOptionClick')

            const expectedActions = store.getActions()
            expect(expectedActions.length).toBe(1)
            expect(expectedActions).toContainEqual({
                type: OPEN_FILTER_SCREEN,
                activeAccordion: undefined
            })
        })
    })

    describe('on clicking FilterTab', () => {
        it('should call openFilterScreen and trackEvent',() => {
            props = {
                selectionObject:{
                    'Budget' : '₹ 5 lakh',
                    'Body Type':'Sedan, Coupe',
                    'Fuel Type':'Diesel, Petrol'
                }
            }
            const wrapper = Wrapper().dive()
            wrapper.instance().trackEvent = jest.fn()

            const editButton = wrapper.find('[data-filter-type="Budget"]')
            editButton.simulate('click', {currentTarget : {
                getAttribute: () => 'Budget'
            }})
            expect(wrapper.instance().trackEvent).toBeCalledWith('EditOptionClick_Budget')

            const expectedActions = store.getActions()
            expect(expectedActions.length).toBe(1)
            expect(expectedActions).toContainEqual({
                type: OPEN_FILTER_SCREEN,
                activeAccordion: 'Budget'
            })
        })
    })


})
