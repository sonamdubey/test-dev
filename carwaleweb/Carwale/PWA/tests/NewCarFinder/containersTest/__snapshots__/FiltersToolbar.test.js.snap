// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for FiltersToolbar component should render snapshot 1`] = `
<div
  className="filters__toolbar-placeholder"
>
  <div
    className="filters-toolbar__content"
  >
    <div
      className="toolbar--relative"
    >
      <div
        className="filters-toolbar-list__content"
      >
        <ul
          className="filter-toolbar__list"
        >
          <li
            className="filter-toolbar-list__item"
            key="Budget"
          >
            <div
              className="btn-tab"
              data-filter-type="Budget"
              onClick={[Function]}
            >
              <span
                className="tab__key"
              >
                Budget
              </span>
              <span
                className="tab__value"
              >
                â‚¹ 5 lakh
              </span>
            </div>
          </li>
          <li
            className="filter-toolbar-list__item"
            key="Body Type"
          >
            <div
              className="btn-tab"
              data-filter-type="Body Type"
              onClick={[Function]}
            >
              <span
                className="tab__key"
              >
                Body Type
              </span>
              <span
                className="tab__value"
              >
                Sedan, Coupe
              </span>
            </div>
          </li>
          <li
            className="filter-toolbar-list__item"
            key="Fuel Type"
          >
            <div
              className="btn-tab"
              data-filter-type="Fuel Type"
              onClick={[Function]}
            >
              <span
                className="tab__key"
              >
                Fuel Type
              </span>
              <span
                className="tab__value"
              >
                Diesel, Petrol
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="filter__edit"
        onClick={[Function]}
      >
        Edit
      </div>
    </div>
  </div>
</div>
`;
